### DISCUSSION TITLE
Alpine V3 & Safari 11 -> Published List of Supported Browsers?

### DISCUSSION THREAD

**Discussion by kylethielk** [alpinejs/alpine#1964] [4. General]
Hi Everyone,

I wanted to post an issue we ran into with Alpine V3 and Safari 11 in case anyone else stumbles on this. We started receiving a trickle of issues being reported after we upgraded to V3 and turns out they were all on Safari 11.

Safari didn't add support for [queueMicrotask](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/queueMicrotask) until v12 and Alpine v3 relies on this method.

The error reported is "Reference Error: Can't find variable: queueMicrotask".

To fix this you can add a polyfill:

```javascript
if (typeof window.queueMicrotask !== "function") {
    window.queueMicrotask = function (callback) {
      Promise.resolve()
        .then(callback)
        .catch(e => setTimeout(() => { throw e; }));
    };
}
```

As a followup question, does Alpine publish a list of supported browsers? This one definitely caught us by surprise.

**Comment by mortenebak** [comment]
I'm using alpine via a cdn - how would I add this polyfill then?

**Reply by danddanddand** [reply]
It's almost no code so either add it to script tag or you can try core-js cdn version but it's probably huge. https://github.com/zloirock/core-js#installation
